packages/server/src/service/alloy/cronjob.ts

```diff
@@ -0,0 +1,38 @@
+import { FirestoreService } from '../firestore'
+import { unfinishedRegistrationStatus, StateInfoForAlloy, recheckRegistration } from './recheck'
+import { processEnvOrThrow } from '../../common'
+import { RichStateInfo } from '../types'
+
+const alloyRecheckInterval: number = +processEnvOrThrow('ALLOY_RECHECK_INTERVAL')
+const firestore = new FirestoreService()
+
+export const crossCheckRegistrationsCronjob = async () => {
+  const now = new Date()
+  const inInterval = now.valueOf() - alloyRecheckInterval
+  const registrations = await firestore.db.collection('StateInfo')
+    .where('alloy.status', 'in', unfinishedRegistrationStatus)
+    .where('alloy.timestamp', '<=', inInterval)
+    .select('id', 'alloy', 'name', 'nameParts', 'address', 'birthdate')
+    .limit(500)
+    .get()
+
+  const updates: Array<Partial<RichStateInfo> & { id: string }> = []
+  for (let i = 0; i < registrations.size; i++) {
+    const data = registrations.docs[i].data() as StateInfoForAlloy
+    const alloy = await recheckRegistration(data)
```

Comment:
Instead of this, do
```
const alloyData = await Promise.all(registrations.docs.map(doc => recheckRegistration (doc.data())))
```
to get alloy data.

https://medium.com/velotio-perspectives/understanding-node-js-async-flows-parallel-serial-waterfall-and-queues-6f9c4badbc17

https://stackoverflow.com/questions/47794063/limit-concurrent-requests-to-api-in-nodejs-along-with-async-await

Reply:
Thanks a lot for this!!